controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/ApplicationVersion.java

```diff
@@ -39,24 +40,22 @@
     public ApplicationVersion(Optional<SourceRevision> source, OptionalLong buildNumber, Optional<String> authorEmail,
                                Optional<Version> compileVersion, Optional<Instant> buildTime, Optional<String> sourceUrl,
                                Optional<String> commit) {
-        Objects.requireNonNull(source, "source cannot be null");
-        Objects.requireNonNull(buildNumber, "buildNumber cannot be null");
-        Objects.requireNonNull(authorEmail, "author cannot be null");
-        if (source.isPresent() != buildNumber.isPresent()) {
-            throw new IllegalArgumentException("both buildNumber and source must be set together");
-        }
-        if (compileVersion.isPresent() != buildTime.isPresent()) {
-            throw new IllegalArgumentException("both compileVersion and buildTime must be set together");
-        }
-        if (buildNumber.isPresent() && buildNumber.getAsLong() <= 0) {
-            throw new IllegalArgumentException("buildNumber must be > 0");
-        }
-        if (authorEmail.isPresent() && ! authorEmail.get().matches("[^@]+@[^@]+")) {
+        if (buildNumber.isEmpty() && (   source.isPresent() || authorEmail.isPresent() || compileVersion.isPresent()
```

Comment:
When can `buildNumber` be empty?

Reply:
`ApplicationVersion.empty`.

Reply:
But is it only used internally or is it possible to get this as response to some job? (I know that at least in the past deployments to dev did not have build number)

Reply:
They explicitly use the `empty` one.

Reply:
:(

Reply:
Why sad face?

Reply:
Would be nice to have something we can use to consistently refer to builds in console